Source/SwiftLintFramework/Rules/Lint/OrphanedDocCommentRule.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+import SourceKittenFramework
+
+public struct OrphanedDocCommentRule: ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "orphaned_doc_comment",
+        name: "Orphaned Doc Comment",
+        description: "A doc comment should be attached to a declaration.",
+        kind: .lint,
+        minSwiftVersion: .fourDotOne,
+        nonTriggeringExamples: [
+            """
+            /// My great property
+            var myGreatProperty: String!
+            """,
+            """
+            //////////////////////////////////////
+            //
+            // Copyright header.
+            //
+            //////////////////////////////////////
+            """
+        ],
+        triggeringExamples: [
+            """
+            ↓/// My great property
+            // Not a doc string
+            var myGreatProperty: String!
+            """
+        ]
+    )
+
+    private static let characterSet = CharacterSet.whitespacesAndNewlines.union(CharacterSet(charactersIn: "/"))
+
+    public func validate(file: SwiftLintFile) -> [StyleViolation] {
+        let docStringsTokens = file.syntaxMap.tokens.filter { token in
+            return token.kind == .docComment || token.kind == .docCommentField
```

Comment:
What kind is a URL in a doc comment?

Reply:
It's `source.lang.swift.syntaxtype.comment.url`:

```swift
/// See
/// https://apple.com
func foo() {}
```

```js
[
  {
    "length" : 8,
    "offset" : 0,
    "type" : "source.lang.swift.syntaxtype.doccomment"
  },
  {
    "length" : 4,
    "offset" : 8,
    "type" : "source.lang.swift.syntaxtype.doccomment"
  },
  {
    "length" : 17,
    "offset" : 12,
    "type" : "source.lang.swift.syntaxtype.comment.url"
  },
  {
    "length" : 1,
    "offset" : 29,
    "type" : "source.lang.swift.syntaxtype.doccomment"
  },
  // ...
]
```

Reply:
Right so shouldn't this check include comment URLs too then?

Reply:
Not really because it's the same syntax kind of a regular comment URL. Since there should be a `source.lang.swift.syntaxtype.doccomment` together, I think it's fine. 

The case where this could lead to false negatives is `/// URL` as the `doccomment` piece would just be `/// ` and then the exception for `/` only lines would avoid the violation to be triggered.

Reply:
Sounds good. Probably a good idea to add a test case to be sure though.

Reply:
Done: https://github.com/realm/SwiftLint/pull/3017 ✅